### PR TITLE
Meaningful stack traces for Elements.

### PR DESCRIPTION
**Pull request by msackman-synthace**
The mutations that we do to the elements mean that the line numbers that come out in the default Go stack traces are not meaningful.
To solve this, we include in each element a map that convers from Go line numbers back to .an line numbers. Using an init block, we then register these.
When recovering from a panic, we then inspect the stack frame (thankfully, Go doesn't unwind this when there's a recover), and try to match this against our registered map of elements and line numbers.
If the first stack frame matches a known element then we treat the whole of the rest of the stack frame as an antha stack frame, and will return an error that renders like this:

```
error: Panic in Element
- [Element Aliquot] repos.antha.com/antha-ninja/elements-westeros/Elements/MVL/Aliquot/element.an:181
       [Go] github.com/antha-lang/elements/vendor/repos.antha.com/elements/Aliquot/element._Steps
            /home/matthew/src/Go_external_1.10.3/src/github.com/antha-lang/elements/vendor/repos.antha.com/elements/Aliquot/element/element.go:175
- [Element Aliquot] repos.antha.com/antha-ninja/elements-westeros/Elements/MVL/Aliquot/element.an:(unknown line)
       [Go] github.com/antha-lang/elements/vendor/repos.antha.com/elements/Aliquot/element.Element.Run
            /home/matthew/src/Go_external_1.10.3/src/github.com/antha-lang/elements/vendor/repos.antha.com/elements/Aliquot/element/element.go:223
- [Element Aliquot] repos.antha.com/antha-ninja/elements-westeros/Elements/MVL/Aliquot/element.an:(unknown line)
       [Go] github.com/antha-lang/elements/vendor/repos.antha.com/elements/Aliquot/element._newRunner.func1
            /home/matthew/src/Go_external_1.10.3/src/github.com/antha-lang/elements/vendor/repos.antha.com/elements/Aliquot/element/element.go:266
- [Go] github.com/antha-lang/antha/inject.(*CheckedRunner).Run
       /home/matthew/src/Go_external_1.10.3/src/github.com/antha-lang/antha/inject/runner.go:60
- [Go] github.com/antha-lang/antha/inject.Call
       /home/matthew/src/Go_external_1.10.3/src/github.com/antha-lang/antha/inject/context.go:77
- [Go] github.com/antha-lang/antha/workflow.(*Workflow).run
       /home/matthew/src/Go_external_1.10.3/src/github.com/antha-lang/antha/workflow/workflow.go:154
- [Go] github.com/antha-lang/antha/workflow.(*Workflow).Run
       /home/matthew/src/Go_external_1.10.3/src/github.com/antha-lang/antha/workflow/workflow.go:202
- [Go] github.com/antha-lang/antha/execute.Run
       /home/matthew/src/Go_external_1.10.3/src/github.com/antha-lang/antha/execute/execute.go:58
- [Go] github.com/antha-lang/antha/cmd/antha/cmd.(*runOpt).Run
       /home/matthew/src/Go_external_1.10.3/src/github.com/antha-lang/antha/cmd/antha/cmd/run.go:240
- [Go] github.com/antha-lang/antha/cmd/antha/cmd.runWorkflow
       /home/matthew/src/Go_external_1.10.3/src/github.com/antha-lang/antha/cmd/antha/cmd/run.go:366
- [Go] github.com/spf13/cobra.(*Command).execute
       /home/matthew/src/Go_external_1.10.3/src/github.com/spf13/cobra/command.go:762
- [Go] github.com/spf13/cobra.(*Command).ExecuteC
       /home/matthew/src/Go_external_1.10.3/src/github.com/spf13/cobra/command.go:852
- [Go] github.com/spf13/cobra.(*Command).Execute
       /home/matthew/src/Go_external_1.10.3/src/github.com/spf13/cobra/command.go:800
- [Go] github.com/antha-lang/antha/cmd/antha/cmd.Execute
       /home/matthew/src/Go_external_1.10.3/src/github.com/antha-lang/antha/cmd/antha/cmd/root.go:60
- [Go] main.run
       /home/matthew/src/Go_external_1.10.3/src/github.com/antha-lang/elements/cmd/antha/main.go:39
- [Go] main.main
       /home/matthew/src/Go_external_1.10.3/src/github.com/antha-lang/elements/cmd/antha/main.go:43
- [Go] runtime.main
       /home/matthew/src/golang/go1.10.3/src/runtime/proc.go:198
- [Go] runtime.goexit
       /home/matthew/src/golang/go1.10.3/src/runtime/asm_amd64.s:2361
```

### CODE REVIEW

**Comment by msackman-synthace** [comment]
Oh I'm completely unsure as to whether inject is the right package to add the new stuff into, so if it should be elsewhere please let me know.

File: inject/linemap.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+	elementMaps = make(map[string]*elementMap)
+}
+
+type elementMap struct {
+	anthaElementPath string
+	elementName      string
+	lineMap          map[int]int
+}
+
+func RegisterLineMap(goElementPath, anElementPath, elementName string, lineMap map[int]int) {
```
Comment (m-i-sadowski) [inline inject/linemap.go]: Can you add comments in standard Gofmt style to public functions please? 

e.g. 
// RegisterLineMap registers a line map 

(but with real content!)

File: inject/linemap.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+	em := &elementMap{
+		anthaElementPath: anElementPath,
+		elementName:      elementName,
+		lineMap:          lineMap,
+	}
+	elementMaps[goElementPath] = em
+}
+
+func ElementStackTrace() string {
```
Comment (m-i-sadowski) [inline inject/linemap.go]: ditto the above

**Comment by msackman-synthace** [comment]
Hold off on this for the moment - it's wrong ;)

**Comment by msackman-synthace** [comment]
Should be good to go now.

**Pull request merged**